### PR TITLE
 Add auto armor equip for kits

### LINKED ISSUE #347: Auto equip armor from kit
There should be an option to automatically equip armor from a kit. It could there could be a separate helmet, chestplate, pants, and boots node to put in an item.

### PR DESCRIPTION
**Pull request by JRoy**
Adds a config option to have player auto-equip armor from kits.

Also adds a performance improvement for all uses of EnumUtil#getAllMatching by switching it to an EnumSet

Closes #347
Closes #3452

### CODE REVIEW

File: Essentials/src/com/earth2me/essentials/Kit.java [hunk 1]
```diff
@@ -191,6 +193,24 @@ public boolean expandItems(final User user, final List<String> items) throws Exc
                     // We pass a null sender here because kits should not do perm checks
                     metaStack.parseStringMeta(null, allowUnsafe, parts, 2, ess);
                 }
+
+                if (autoEquip) {
+                    ItemStack stack = metaStack.getItemStack();
+                    Material material = stack.getType();
+                    if (MaterialUtil.isHelmet(material) && isEmptyStack(user.getBase().getInventory().getHelmet())) {
+                        user.getBase().getInventory().setHelmet(stack);
```
Comment (montlikadani) [inline Essentials/src/com/earth2me/essentials/Kit.java]: Please make a new field for `user.getBase().getInventory()`

File: Essentials/src/com/earth2me/essentials/utils/MaterialUtil.java [hunk 2]
```diff
@@ -25,9 +25,26 @@
     private static final Set<Material> SIGN_POSTS;
     private static final Set<Material> WALL_SIGNS;
 
+    private static final Set<Material> HELMETS;
+    private static final Set<Material> CHESTPLATES;
+    private static final Set<Material> LEGGINGS;
+    private static final Set<Material> BOOTS;
```
Comment (montlikadani) [inline Essentials/src/com/earth2me/essentials/utils/MaterialUtil.java]: I would rather separate these with commas into one field.
```suggestion
    private static final Set<Material> HELMETS, CHESTPLATES, LEGGINGS, BOOTS;
```

Comment (JRoy) [reply Essentials/src/com/earth2me/essentials/utils/MaterialUtil.java]: Doesn't follow our codestyle, let alone the codestyle of this file.

**Comment by JRoy** [comment]
[Kit Armor Auto Equip](https://trello.com/c/jP1X3woH/6-kit-armor-auto-equip)

**Review by mdcfe — APPROVED** [review]

**Pull request merged**